✨ TJ-13 알림창 컴포넌트 생성

yuyoni opened the pull request:
## 주요 변경 사항

- 모달이 보이는지 안보이는지를 나타내는 상태, 모달 상태를 조작할 수 있는 함수,  알림목록을 prop으로 받는 알림창 NotificationModal 컴포넌트 생성
- 알림창 내부의 알림 Notification 컴포넌트 생성
- 경과시간을 계산해서 반환하는 getElaspedTime 유틸함수 생성
- 시작 시간과 근무 시간을 매개변수로 받아 문자열로 실제 근무 시간을 반환하는 formatTimeRange 유틸함수 생성

## 관련 스크린샷

- PC 및 태블릿 : X 버튼 없음<br>
<img width="300" alt="image" src="https://github.com/the-julge/the-julge/assets/127701092/d9be9775-3c3a-4565-923c-611c456fc3fe"><br><br>
- 모바일 : 화면 전체를 차지하므로 X 버튼 있음<br>
<img src="https://github.com/the-julge/the-julge/assets/127701092/8e3f2158-19f9-4efd-8216-c8e98267c122" width="300px" />


## 리뷰어에게

- 특정 사용자의 알림목록을 조회하는 API로부터 받아온 응답 중 items 프로퍼티 값을 알림목록으로 넘겨줄 수 있도록 했습니다
![image](https://github.com/the-julge/the-julge/assets/127701092/4089a2ea-ae21-47e6-911b-5124926d525c)
- PC 및 태블릿에서는 X버튼이 없어서 모달을 사용하는 부모 컴포넌트에서 모달외부 영역을 누르거나 알림모달 조작버튼을 누르면 닫히도록 따로 작성해줘야합니다..!

yuyoni committed:
✨ feat: 알림 데이터를 props로 받는 NotificationModal 컴포넌트 생성

- 알림창 NotificationModal 컴포넌트 생성
- 알림창 내부의 알림 Notification 컴포넌트 생성
- 경과시간을 계산해서 반환하는 getElaspedTime 유틸함수 생성
- 시작 시간과 근무 시간을 매개변수로 받아 문자열로 실제 근무 시간을 반환하는 formatTimeRange 유틸함수 생성

yuyoni committed:
🐛 fix: 알림 상태 (승인, 거절)을 나타내는 svg 별도 파일로 분리하지 않고 컴포넌트 내에서 조작하도록 수정

yuyoni committed:
♻️ refactor: 날짜 형식에서 불필요한 부분 제거

yuyoni committed:
✨ feat: 날짜 형식 수정

- .을 이용한 날짜 표시에서 피그마에 나온 하이픈을 이용해 구분한 형식으로 변경

yuyoni committed:
🚚 rename: public 폴더 내에 images 하위 폴더 생성

midcondria commented on components/NotificationModal/Notification.tsx:
```diff
@@ -0,0 +1,57 @@
+}: Notification) {
+  return (
+    <Wrapper>
+      <svg
+        xmlns="http://www.w3.org/2000/svg"
+        width="5"
+        height="5"
+        viewBox="0 0 5 5"
+        fill="none"
+      >
+        <circle
+          cx="2.5"
+          cy="2.5"
+          r="2.5"
+          fill={
+            result === "accepted"
+              ? "var(--The-julge-blue-20, #0080FF)"
+              : "var(--The-julge-red, #EC5A46)"
+          }
+        />
+      </svg>
```
개인적으로는 요런거는 이렇게 보면 무슨 이미지인지 감이 안잡혀서
하단에 컴포넌트로 뽑아서 이름 지어주는것도 괜찮지 않을까 싶어요.

yuyoni replied:
앗 멘토님이 전에 Image 태그의 최적화 기능 사용안할거면 그냥 svg 태그로 써도 괜찮을 것 같다고 해서 써보긴했는데
가독성 떨어지긴하는 것 같군요.. ㅎㅎ  파일 분리하도록 하겠습니다!

midcondria reviewed (approved):
고생하셨습니다 :thumbsup:

yuyoni committed:
♻️ refactor: type 키워드에서 interface 키워드로 수정

-  다른 파일에서 객체 타입 선언할 때 interface를 사용해서, 코드를 일관적으로 작성하기 위해 type 키워드를 interface 키워드로 수정합니다

soyxxn reviewed (approved):
우와아 수고하셨습니다 🥰❤️❤️ !!!
